llvm/unittests/Support/KnownBitsTest.cpp

```diff
@@ -653,8 +656,10 @@ TEST(KnownBitsTest, GetMinMaxVal) {
       Min = APIntOps::umin(Min, N);
       Max = APIntOps::umax(Max, N);
     });
-    EXPECT_EQ(Min, Known.getMinValue());
-    EXPECT_EQ(Max, Known.getMaxValue());
+    if (!Known.hasConflict()) {
```

Comment:
If there's a conflict you could immediately "continue" at the top of the loop to save some time. Same in the next three functions below.

Reply:
But in this case you are not calling `Known.anything()` before checking for conflict and bailing out!

Reply:
Hm, yes, I guess we should store Known.getMinValue() etc in a variable first, before the check. Or do people think there is not much value in checking all the functions with conflict inputs?

Reply:
> Or do people think there is not much value in checking all the functions with conflict inputs?

I'm not sure how much value there is right now.

At some point we should check that KnownBits functions return conflict when they ought to, and part of that would definitely involve testing them on conflict inputs.

Reply:
These particular APIs don't return KnownBits, so they don't really have any specific behavior they should exhibit for conflict inputs beyond "don't assert".